python/pyspark/errors/utils.py

```diff
@@ -31,21 +31,41 @@
     Type,
     Optional,
     Union,
-    TYPE_CHECKING,
     overload,
     cast,
 )
 import pyspark
 from pyspark.errors.error_classes import ERROR_CLASSES_MAP
 
-if TYPE_CHECKING:
-    from pyspark.sql import SparkSession
-
 T = TypeVar("T")
 FuncT = TypeVar("FuncT", bound=Callable[..., Any])
 
 _current_origin = threading.local()
 
+# Providing DataFrame debugging options to reduce performance slowdown.
+# Default is True.
+_enable_debugging_cache = None
+
+
+def is_debugging_enabled() -> bool:
+    global _enable_debugging_cache
+
+    if _enable_debugging_cache is None:
+        from pyspark.sql import SparkSession
+
+        spark = SparkSession.getActiveSession()
+        if spark is not None:
+            _enable_debugging_cache = (
+                spark.conf.get(
+                    "spark.sql.dataFrameQueryContext.enabled", "true"  # type: ignore[union-attr]
```

Comment:
`spark.sql.dataFrameQueryContext.enabled` has to be StaticSQLConf

Reply:
I see. Should we update the existing conf to static conf or should we add a new one for PySpark specific?

The existing conf here is defined as `SQLConf` below:

https://github.com/apache/spark/blob/e55511c98e14115dd023f9014372fc7f23d9ef6c/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L5223-L5230

Reply:
I think so

Reply:
Also cc @vladimirg-db